src/printer.js

```diff
@@ -3431,7 +3431,16 @@ function printJSXChildren(path, options, print, jsxWhitespace) {
           if (beginSpace) {
             children.push(jsxWhitespace);
           } else {
-            children.push(softline);
+            // Ideally this would be a `softline` to allow a break between
+            // tags and text.
+            // Unfortunately Facebook have a custom translation pipeline
+            // (https://github.com/prettier/prettier/issues/1581#issuecomment-300975032)
+            // that uses the JSX syntax, but does not follow the JSX whitespace
```

Comment:
I would call this "the React whitespace rules" since JSX does not seem to have whitespace rules.